producer/src/block/block.go

```diff
@@ -13,24 +13,37 @@ type Block struct {
 	PreviousHash   []byte
 	MerkleRootHash []byte
 	Data           [][]byte
+	DataLen        uint16
 }
 
 // Produces a block based on the struct provided
-func (b *Block) Serialize() []byte {
-	// allocates space for the known variables
-	serializedBlock := make([]byte, 20)
+func (b *Block) Serialize() []byte { // Vineet
+	//calculate the total length beforehand, to prevent unneccessary appends
+	//NOTE: 32 bit ints are used to hold lengths; unsigned 16 bit int is used for the length of Data
+	bLen := 4 + 8 + 8 + 32 + 32 + 2 //size of all fixed size fields
```

Comment:
Can we make this a static number and just put it straight into the function call to reduce operation cost?

Reply:
Done